chainer/gradient_check.py

```diff
@@ -279,29 +284,47 @@ def check_backward(func, x_data, y_grad, params=(),
     params_grad = [param.grad for param in params]
 
     xp = cuda.get_array_module(*xs)
-    one = xp.array(1., dtype)
+    xs_directions = [
+        xp.random.normal(size=x.shape).astype(dtype) for x in casted_xs]
+    param_directions = [
+        xp.random.normal(size=x.shape).astype(dtype) for x in param_data]
+    # Use unit vector
+    norm = math.sqrt(
+        sum([xp.square(x).sum() for x in xs_directions]) +
+        sum([xp.square(x).sum() for x in param_directions]))
+    if norm > 0:
+        scale = 1. / norm
+    else:
+        scale = 1.
```

Comment:
To handle `norm == 0` case, we should re-sample the direction, because it is zero-vector.

Reply:
`norm==0` happens when all inputs are skipped. But now i think we can skip such test case...